crates/autopilot/src/run_loop.rs

```diff
@@ -649,7 +633,16 @@ impl RunLoop {
                 })?;
                 onchain_task.await
             }
+        }?;
+        if solved.solver() != solver {
```

Comment:
That `wait_for_settlement_transaction` can return when we see the tx from a different solver is a problem. This would mean in a world where multiple solvers submit solutions we'd very often not wait for the correct tx to appear on chain an then log this error.
Instead `wait_for_settlement_transaction` should take the expected `solver` address and only return if that solver actually submitted something.

Reply:
Actually I planned to solve it differently and not use `solver` field reported by solvers (as I think it's not a type of data that core protocol functionality should rely on). Instead, for multiple winners, I planned to just wait for any N number of detected settlements if N settle calls are issues.

But then, if we use `solver` field to reward participants, solvers are somewhat incentivized to be honest around this field so I also think you suggestion is valid for now and makes the multiple settlements functionality easier to implement.

Reply:
Fixed